src/worker.js

```diff
@@ -30,9 +30,10 @@ if (ENVIRONMENT_IS_NODE) {
     self: global,
     require: require,
     Module: Module,
-    location: {
-      href: __filename
-    },
+#if !EXPORT_ES6
+    __filename: __filename,
```

Comment:
Why do these variables need to be propagated to global object?

Reply:
IIUC, these global variables are propagated from the main thread to the worker. I haven't investigated why `__filename` and `__dirname` are `undefined` in the worker when not doing so.

Reply:
(I explicitly didn't this for `EXPORT_ES6`, as that would be redundant)